### PR TITLE
Update to Laravel 5.1

### PR DESCRIPTION
**Pull request by chaosfinity**
Rebase #22 to master

As far as continuing support for Laravel 4 I can work on that. It would be a provider that delegates to the correct provider based on version.

For your concern about being able to change the response by making a new event listener on Laravel 5 you can do that with middleware just watching for the response types that are returned by the controller. If developers still want to catch exceptions instead of using middleware to convert the response there could be a config option to throw exceptions as before. What do you think?

### CODE REVIEW

**Comment by makasim** [comment]
> If developers still want to catch exceptions instead of using middleware to convert the response there could be a config option to throw exceptions as before. What do you think?

That's ok to keep it simple, So let's keep it as it is now and see later

**Comment by makasim** [comment]
> As far as continuing support for Laravel 4 I can work on that. It would be a provider that delegates to the correct provider based on version.

@chaosfinity It would be great if you can do in a BC way.

**Comment by chaosfinity** [comment]
The service provider now delegates to a version specific provider.

To allow the v4 provider to throw reply exceptions and the v5 to convert the reply directly I did have to add a configuration option. I did not document the config option so the path can be changed if you dont like the "payum-laravel-package.settings.throwReplyExceptions" path.

File: composer.json [hunk 1]
```diff
@@ -35,9 +35,9 @@
     ],
     "require": {
         "php": ">=5.3.2",
-        "illuminate/container": "~4.0",
-        "illuminate/routing": "~4.0",
-        "illuminate/support": "~4.0",
+        "illuminate/container": "~5.1",
+        "illuminate/routing": "~5.1",
+        "illuminate/support": "~5.1",
```
Comment (makasim) [inline composer.json]: @chaosfinity could you change this to "~4.0|~5.1"?

File: src/Payum/LaravelPackage/Controller/AuthorizeController.php [hunk 2]
```diff
@@ -16,7 +16,10 @@ public function doAction($payum_token)
 
         $gateway = $this->getPayum()->getGateway($token->getGatewayName());
 
-        $gateway->execute(new Authorize($token));
+        $response = $this->convertReply($gateway->execute(new Authorize($token), true));
+
+        if($response)
+            return $response;
```
Comment (makasim) [inline src/Payum/LaravelPackage/Controller/AuthorizeController.php]: Please follwo psr1\2 coding standards

```php
<?php
if($response) {
    return $response;
}
```

**Comment by makasim** [comment]
Thanks.

**Pull request merged**